core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala

```diff
@@ -175,23 +176,23 @@ class ProducerFailureHandlingTest extends KafkaServerTestHarness {
   }
 
   /**
-   *  The send call with invalid partition id should throw KafkaException caused by IllegalArgumentException
-   */
+    * Send with invalid partition id should throw KafkaException when partition is higher than the
+    * upper bound of partitions and IllegalArgumentException when partition is negative
```

Comment:
`PlaintextProducerSendTest` is the right test, `ProducerTest` is deprecated. Yes, I noticed the one that uses mocking, but that doesn't verify that one refresh is actually enough after we add partitions (which is the actual goal of this PR).